### DISCUSSION TITLE
social card / mkdocs brew for windows

### DISCUSSION THREAD

**Discussion by H3x37** [squidfunk/mkdocs-material#6242] [General]
i have a issue to **install cairo freetype libffi libjpeg libpng zlib** on my windows i search so much install wsl and the kernal etc ... but i need to install that packge on my windows for social card on my mkdocs any guide will be nice thank/s alote

**Comment by alexvoss** [comment]
I have looked at installing these libraries on Windows and it is much more difficult than it should be. Iterating over all the dependencies and installing them manually should work but will be tedious and difficult to repeat. 

What I was looking for was a solution that would work like things do on Linux or MacOS (with [Homebrew](https://brew.sh/)). I tried [MSYS2](https://www.msys2.org/) but ran into various issues and gave up.

My best advice is to install the [Linux subsystem](https://learn.microsoft.com/en-us/windows/wsl/install) and use that for the builds. You can still use your Windows apps for editing and your Windows browser(s) for viewing the results. It seems you have already explored this. If you want us to help then we will need more info about just what specific issues you ran into.

**Comment by kamilkrzyskow** [comment]
I don't know about the Social Cards plugin from Insiders, but if you're using the free community version then as written in the documentation install the precompiled GTK Runtime:
https://squidfunk.github.io/mkdocs-material/plugins/requirements/image-processing/#cairo-graphics-windows

1. https://github.com/tschoonj/GTK-for-Windows-Runtime-Environment-Installer/releases/download/2022-01-04/gtk3-runtime-3.24.31-2022-01-04-ts-win64.exe
2. Afterwards make sure that the `<install-directory>\bin` was added to either the user `Path` or system `Path`, either is fine.
    You can press `<WIN>` + `<R>` and type `SystemPropertiesAdvanced` then follow this guide to see if has been added during installation, if not input it yourself. 
    https://www.howtogeek.com/787217/how-to-edit-environment-variables-on-windows-10-or-11/
3. Don't forget to first close the windows with OK to save changes, it's not saving it immediately after you input the text.
4. Afterwards make sure that you restart the terminal / console window where you run `mkdocs serve`, also if you're running a Text Editor which has an integrated console (ex. Visual Studio Code) then restart the whole Text Editor, PC restart shouldn't be required.
5. Enjoy

<details>
<summary>Image for reference:
</summary>

![image](https://github.com/squidfunk/mkdocs-material/assets/34622465/2d3eef10-29f3-430a-82b5-8ffce5370be5)

</details>

**Comment by squidfunk** [comment]
As an alternative, our [Docker image](https://hub.docker.com/r/squidfunk/mkdocs-material/) comes with all dependencies pre-installed. If you're on Insiders, you'd need to build that first, but that's likely much simpler than installing all dependencies on Windows.
